wgpu/src/lib.rs

```diff
@@ -2124,6 +2124,31 @@ impl Device {
         }
     }
 
+    /// Creates a shader module from either SPIR-V or WGSL source code without uniformity checks.
+    ///
+    /// # Safety
+    /// In contrast with [`create_shader_module`](Self::create_shader_module) this function
+    /// creates a shader module without unifromity checks which allows shaders to perform
+    /// operations which can lead to undefined behavior like data races, thus it's
+    /// the caller responsibility to pass a shader which doesn't perform any of these
+    /// operations.
+    ///
+    /// This is equivalent to `create_shader_module` effect on web.
+    pub unsafe fn create_shader_module_non_uniform(
```

Comment:
I think we should combine this function with the `create_shader_module_unchecked` into a function that just takes an explicilt `wgt::ShaderBoundChecks` argument.

Reply:
I'd avoided doing this because I couldn't find a case where `wgpu_types` types were exposed in the public API of wgpu.

If that's permissible then I'll add that.

Reply:
Oh, it's called `wgt`, and so searching for `wgpu_types` in the `wgpu` folder gives no useful results.

That's not very helpful